crates/driver/src/domain/mempools.rs

```diff
@@ -77,6 +77,12 @@ pub enum RevertProtection {
     Disabled,
 }
 
-#[derive(Debug, Error)]
-#[error("none of the submission strategies successfully submitted the solution")]
-pub struct AllFailed;
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Mined reverted transaction: {0:?}")]
+    Revert(eth::TxId),
+    #[error("Simulation started reverting during submission")]
+    SimulationRevert,
```

Comment:
Eventually it would be nice if we could put the current block into the `SimulationRevert`.
Doesn't have to happen in this PR but that might be a good time to base the periodic simulations on the `current_block_stream` to only simulate when we actually saw a new block.

Reply:
Tracked https://github.com/cowprotocol/services/issues/2017